src/io/der.rs

```diff
@@ -308,4 +308,46 @@ mod tests {
             });
         }
     }
+
+    fn bytes_reader(bytes: &[u8]) -> untrusted::Reader {
+        return untrusted::Reader::new(untrusted::Input::from(bytes));
+    }
+
+    #[test]
+    fn test_bit_string_with_no_unused_bits() {
```

Comment:
Independently of the idea of sharing the test logic between the fuzzer and the known-answer tests, it is really important that tests in *ring* are written as parameterized tests with minimal boilerplate.
   
```
    use bit_string_with_no_unused_bits_tests::TestCase;
    const TEST_CASES: &[TestCase] = &[
             // the comment for the test case.
            TestCase { 
                input: &[0x01, 0x01, 0xff],
                result: Err(error::Unspecified),
            },
            ...
        ];
        TEST_CASES.for_each(|TestCase { input, result, .. }| {
           let mut reader = ...;
           assert_eq!(bit_string_with_no_unused_bits(&mut reader), result);
        });
}